persistence/src/main/java/de/cyface/persistence/PersistenceLayer.java

```diff
@@ -1023,6 +1028,25 @@ public void setDistance(final long measurementIdentifier, final double newDistan
         updateMeasurement(measurementIdentifier, values);
     }
 
+    /**
+     * Updates the entries relevant for {@link Measurement#getAverageSpeed()} of the currently captured {@link Measurement}.
+     * <p>
+     * <b>ATTENTION:</b> This should not be used by SDK implementing apps.
+     *
+     * @param measurementIdentifier The id of the {@link Measurement} to be updated
+     * @param newSpeedSum The new sum of all valid speed samples.
+     * @param newSpeedSamples The new number of valid speed samples.
+     * @throws NoSuchMeasurementException if there was no {@code Measurement} with the id
+     *             {@param measurementIdentifier}.
+     */
+    public void setAverageSpeed(final long measurementIdentifier, final double newSpeedSum, final int newSpeedSamples)
+            throws NoSuchMeasurementException {
+        final ContentValues values = new ContentValues();
+        values.put(COLUMN_SPEED_SUM, newSpeedSum);
+        values.put(COLUMN_SPEED_COUNTER, newSpeedSamples);
+        updateMeasurement(measurementIdentifier, values);
+    }
```

Comment:
My understanding of the feature request was:
- the average speed per measurement should be calculated
- i.e. when we pause the first track of the measurement and resume, the speed samples of all sub-tracks are used to calculate the average speed
- for this to work, we cannot save "averageSpeed" to the database but have to save the speed sum and counter.